modules/vpc-endpoints/main.tf

```diff
@@ -0,0 +1,47 @@
+locals {
+  enabled = module.this.enabled && (length(var.gateway_vpc_endpoints) > 0 || length(var.interface_vpc_endpoints) > 0)
+}
+
+data "aws_vpc" "vpc" {
+  count = local.enabled ? 1 : 0
+  id    = var.vpc_id
+}
+
+data "aws_vpc_endpoint_service" "gateway_endpoint_service" {
+  for_each     = var.gateway_vpc_endpoints
```

Comment:
what if `module.this.enabled == false`, but `var.gateway_vpc_endpoints` or `var.interface_vpc_endpoints` is provided?
the data sources and resources with `for_each` will still be read and created.

You can use something like

```
for_each     = local.enabled ? var.gateway_vpc_endpoints : []
```